gax/src/main/java/com/google/api/gax/retrying/RetryAlgorithm.java

```diff
@@ -43,19 +43,41 @@
  * @param <ResponseT> response type
  */
 public class RetryAlgorithm<ResponseT> {
-  private final ResultRetryAlgorithm<ResponseT> resultAlgorithm;
-  private final TimedRetryAlgorithm timedAlgorithm;
+  private final ResultRetryAlgorithmWithContext<ResponseT> resultAlgorithm;
+  private final TimedRetryAlgorithmWithContext timedAlgorithm;
 
   /**
    * Creates a new retry algorithm instance, which uses thrown exception or returned response and
    * timed algorithms to make a decision. The result algorithm has higher priority than the timed
    * algorithm.
    *
+   * <p>Instances that are created using this constructor will ignore the {@link RetryingContext}
```

Comment:
We could consider deprecating this constructor to indicate that the one that will respect the `RetryingContext` should be used.